app/services/save_practice_service.rb

```diff
@@ -161,33 +166,40 @@ def update_category_practices
     practice_category_practices = @practice.category_practices
     practice_categories = @practice.categories
-    if !@current_endpoint.nil? && @current_endpoint.downcase == "introduction"
+    if @current_endpoint.present? && @current_endpoint.downcase == "introduction"
       covid_category_notifications(category_params, practice_categories)
     end
+
     if category_params.present?
       category_attribute_params = @practice_params[:categories_attributes]
       cat_keys = category_params.keys
 
       cat_keys.each do |key|
-        if practice_categories.ids.exclude?(key.to_i)
+        if practice_categories.ids.exclude?(key.to_i) && !key.include?('other')
           practice_category_practices.find_or_create_by(category_id: key.to_i)
         end
       end
 
-      other_cat_id = Category.find_by(name: 'Other')&.id
-
-      if other_cat_id.present? && cat_keys.include?(other_cat_id.to_s)
-        categories_to_process = category_attribute_params.values.map { |param| {id: param[:id], name: param[:name], _destroy: param[:_destroy]} }
-        # If Other was checked, create a new category with is_other true and create a category_practice linking to the new category
-        categories_to_process.each do |category|
-          unless category[:name] == ""
-            if category[:_destroy] == 'false' && category[:id].nil?
-              cate = Category.create(name: category[:name], is_other: true)
-              practice_category_practices.create(category_id: cate.id)
-            elsif category[:_destroy] == 'false' && category[:id].present?
-              practice_categories.find_by(id: category[:id].to_i).update_attributes(name: category[:name])
-            elsif category[:_destroy] == 'true' && category[:id].present?
-              practice_category_practices.where(category_id: category[:id]).destroy_all
+      other_parent_cats = []
+      cat_keys.each { |ck| if ck.include?('other') then other_parent_cats << Category.where('lower(name) = ?', ck.split('-').pop.downcase).first end }
+
+      if other_parent_cats.present?
+        category_attribute_params.values.each do |category|
+          # If Other was checked, create a new category with is_other true and create a category_practice linking to the new category
+          id = category[:id]
+          destroy = category[:_destroy]
+          name = category[:name]
+          parent_cat_id_param = category[:parent_category_id]
+          # if the 'other' category has not yet been created, the :parent_category_id will be a string, so we need to find the corresponding parent category
+          parent_cat_id = parent_cat_id_param.to_i === 0 ? Category.where('lower(name) = ?', parent_cat_id_param.downcase).first.id : parent_cat_id_param
+          unless name == ""
+            if destroy == 'false' && id.blank?
+              cate = Category.find_or_create_by!(name: name.strip.downcase, is_other: true, parent_category_id: parent_cat_id)
```

Comment:
I think we should do the find first  `Category.find_by('lower(name) = ?', name.strip.downcase, is_other: true, parent_category_id: parent_cat_id)` and then create it if it doesn't exist

Right now if we had an existing other category with the same parent with the name `Foobar` we would create the other category if the user submitted any version of 'Foobar'.

Reply:
Addressed